Undefined `ocaml_terminal_get_sigwinch` on macos

The newest progress version fails at link time with the following error:

```
Undefined symbols for architecture x86_64:
  "_ocaml_terminal_get_sigwinch", referenced from:
      _camlTerminal__entry in terminal.a(terminal.o)
      _camlTerminal__gc_roots in terminal.a(terminal.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/src/terminal/terminal_stubs.c
+++ b/src/terminal/terminal_stubs.c
@@ -11,7 +11,7 @@
 // Detect platform
 #if defined(_WIN32) || defined (_WIN64)
 #define OCAML_TERMINAL_WINDOWS
-#elif defined(__unix__) || defined(__unix)
+#elif defined(__unix__) || defined(__unix) || (defined(__APPLE__) && defined(__MACH__))
 #include <unistd.h>
 #if defined(_POSIX_VERSION)
 #define OCAML_TERMINAL_POSIX

